mlflow/models/evaluation/base.py

```diff
@@ -1539,11 +1553,17 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
     ) = _normalize_evaluators_and_evaluator_config_args(evaluators, evaluator_config)
 
     with _start_run_or_reuse_active_run() as run_id:
+        # Convert data to `mlflow.data.dataset.Dataset`.
+        data = _convert_data_to_mlflow_dataset(
+            data=data,
+            targets=targets,
+            dataset_path=dataset_path,
+        )
         from mlflow.data.pyfunc_dataset_mixin import PyFuncConvertibleDatasetMixin
 
-        if isinstance(data, Dataset) and issubclass(data.__class__, PyFuncConvertibleDatasetMixin):
+        if issubclass(data.__class__, PyFuncConvertibleDatasetMixin):
```

Comment:
Do we think it would ever go into the else block:
```
else:
            dataset = EvaluationDataset(
                data,
                targets=targets,
                path=dataset_path,
                feature_names=feature_names,
            )
            Is creating the `EvaluationDataset` needed anymore if we always have created `Dataset` now? 
```

Reply:
Good question - at the if check, `data` will be guaranteed as a `Dataset`, but I am not sure if it is possible that `issubclass(data.__class__, PyFuncConvertibleDatasetMixin)` check fails. I am keeping the `else` branch to avoid regression, but idk if it is truly necessary.